spec/controllers/automated_tests_controller_spec.rb

```diff
@@ -1,6 +1,78 @@
 describe AutomatedTestsController do
   let(:assignment) { create :assignment }
   let(:admin) { create :admin }
+  context 'GET download_specs' do
+    context 'when the file exists' do
+      let(:content) { '{"a":1}' }
+      before :each do
+        FileUtils.mkdir_p assignment.autotest_path
+        File.write(assignment.autotest_settings_file, content)
+        get_as admin, :download_specs, params: { assignment_id: assignment.id }
+      end
+      it 'should download a file containing the content' do
+        expect(response.body).to eq content
+      end
+      it 'should respond with a success' do
+        expect(response.status).to eq 200
+      end
+    end
+    context 'when the file does not exist' do
+      before :each do
+        FileUtils.rm_f(assignment.autotest_settings_file)
+        get_as admin, :download_specs, params: { assignment_id: assignment.id }
+      end
+      it 'should download a file with an empty hash' do
+        expect(response.body).to eq '{}'
+      end
+      it 'should respond with a success' do
+        expect(response.status).to eq 200
+      end
+    end
+  end
+  context 'POST upload_specs' do
+    before :each do
+      FileUtils.mkdir_p assignment.autotest_path
```

Comment:
This part is interesting! I did some digging and it looks like the only (?) place that creates this directory is under `AutomatedTestsController#manage`. This strikes me as not great in general, and especially now since you're adding to the API.

One possibility is to create the directory when we create the assignment. Another possibility is to do an existence check in the `Assignment#autotest_path` method itself. Or you might have other suggestions.